packages/flutter/lib/src/cupertino/refresh.dart

```diff
@@ -143,9 +143,22 @@ class _RenderCupertinoSliverRefresh extends RenderSliver
       parentUsesSize: true,
     );
     if (active) {
+      // The _CupertinoActivityIndicatorPainter performs an internal translation so that
+      // its painting is always vertically centered based on its height. However, the
+      // requirement here is that the top of the activity indicator (not the center) is
+      // pinned to the top of the sliver, so we need to account for the internal adjustment
+      // here taking into account that the child's height will change as the user drags down.
+      final double internalPainterDelta = _activityIndicatorRadius - (min(child.size.height / 2.0, _activityIndicatorRadius));
+
+      // The standard iOS pull-to-refresh layout includes a margin above the activity indicator.
+      // A subtle consequence of that is that in the very early stages of the drag, a portion
+      // the first tick is actually drawn over the top of the content being dragged down. To
+      // achieve the same effect, we apply the margin to the paintOrigin.
+      const double topMargin = 16.0;
+
       geometry = SliverGeometry(
         scrollExtent: layoutExtent,
-        paintOrigin: -overscrolledExtent - constraints.scrollOffset,
+        paintOrigin: topMargin + internalPainterDelta - overscrolledExtent - constraints.scrollOffset,
```

Comment:
Sorry for the delay. Thank you for putting detailed comments in the code!

Is it possible to add a padding to the activity indicator, instead of applying the `paintOrigin` change, and achieve the same thing? My concern is it might hurt the customizability of this widget, if the sliver has a hard-coded top margin.

Reply:
Also, here's a gif showing the native behavior:
![refresh-control](https://user-images.githubusercontent.com/31859944/84831507-485fe280-afe0-11ea-8576-d216e106f4d3.gif)

The native refresh control seems to become fully on screen (in terms of extent) as soon as the drag begins, and starts to fade in as the user keeps dragging the content down.

When the refresh control becomes active (`armed` or `refresh`), it stays at the top unless the user drags the content up far enough, at which point it simply disappears. It reappears when the content goes back down.

Reply:
I think we can remove the height constraint from the `BoxConstriants` we apply one the `child`: https://github.com/flutter/flutter/blob/a0d8fdf393816cbd222c98e1e15ce3602cc4078e/packages/flutter/lib/src/cupertino/refresh.dart#L136-L143, but that will be a hard breaking change, as it breaks every custom refresh indicator that doesn't have an intrinsic height.